### ISSUE TITLE
[MNG-6161] Dependencies' management via <scope>import</scope> should take precedence over inherited definitions

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#7281]
**[Daniel McGreal](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=djmcgreal)** opened **[MNG-6161](https://issues.apache.org/jira/browse/MNG-6161?redirect=false)** and commented

Hi, I would like to suggest an improvement to the way in which \<scope>import\</scope> dependencies are handled.

Currently, if I have four projects:
* 'version' which sets a dependency on X with a version of Y, and is a parent pom
* 'depender' which inherits from 'version' and declares a dependency on X
* 'provider' which manages X to scope provided
* 'last' which depends on 'depender' and imports 'provider's managed dependencies
  then the 'provider' definition is ignored and 'version's is chosen.

The use case I'm trying to achieve is where some dependencies are excluded from a plugin's activities, and using \<scope>provided\</scope> is the best way for me to achieve this. Using \<scope>import\</scope> to override hierarchically defined dependency management would be a nice way of organising this in a way that supports the multi-tired architecture my real use case would require.

In short, it feels to me that a \<dependencyManagement> definition with \<scope>import\</scope> should be 'closer' to the current bundle's dependency resolution than its parents.

Thanks for your consideration,
Dan.

---

**Affects:** 3.3.9

**Issue Links:**
- [MNG-5971](https://issues.apache.org/jira/browse/MNG-5971) Imported dependencies should be available to inheritance processing
 (_**"duplicates"**_)

1 votes, 2 watchers
